### ISSUE TITLE
Question about the collect_fn in fine-tuning demo

### ISSUE THREAD

**Issue by Li-Qingyun** [huggingface/smollm#70]
Hi, it seems that the collate_fn in [fine-tuning demo](https://github.com/huggingface/smollm/blob/main/vision/finetuning/SmolVLM2_Video_FT.ipynb) only ignore ths image tokens and learn all text tokens.

```
    labels = pad_sequence(
        [inst["input_ids"].squeeze(0).clone() for inst in instances],
        batch_first=True,
        padding_value=-100
    )

    labels[labels == image_token_id] = -100
```

I consider that only the labels where correspond to response section should be supervised. But here both prompt messages and response messages are learned.

What do the authors think of this problem? Is there a convenient manner to implement the well-used `only learn reponse` scheme with apply_chat_template?
